install/management/appliance.go

```diff
@@ -0,0 +1,395 @@
+
+	var vicadmintlsargs string
+	if conf.CertPEM != "" && conf.KeyPEM != "" {
+		spec.ExtraConfig = append(
+			spec.ExtraConfig,
+			&types.OptionValue{Key: "guestinfo.vch/etc/pki/tls/certs/vic-host-cert.pem", Value: conf.CertPEM},
+		)
+		spec.ExtraConfig = append(
+			spec.ExtraConfig,
+			&types.OptionValue{Key: "guestinfo.vch/etc/pki/tls/certs/vic-host-key.pem", Value: conf.KeyPEM},
+		)
+		d.dockertlsargs = "-TLS -tls-certificate=/etc/pki/tls/certs/vic-host-cert.pem -tls-key=/etc/pki/tls/certs/vic-host-key.pem"
+		vicadmintlsargs = " -hostcert=/etc/pki/tls/certs/vic-host-cert.pem -hostkey=/etc/pki/tls/certs/vic-host-key.pem"
+		files = fmt.Sprintf("%s /etc/pki/tls/certs/vic-host-cert.pem /etc/pki/tls/certs/vic-host-key.pem", files)
+		d.DockerPort = "2376"
+	} else {
+		d.DockerPort = "2375"
+	}
+	spec.ExtraConfig = append(spec.ExtraConfig, &types.OptionValue{Key: "guestinfo.vch/files", Value: files})
+	spec.ExtraConfig = append(spec.ExtraConfig, &types.OptionValue{Key: "guestinfo.vch/sbin/docker-engine-server",
+		Value: fmt.Sprintf("-serveraddr=0.0.0.0 -port=%s -port-layer-port=8080 %s", d.DockerPort, d.dockertlsargs)})
+	spec.ExtraConfig = append(spec.ExtraConfig, &types.OptionValue{Key: "guestinfo.vch/sbin/vicadmin",
+		Value: fmt.Sprintf("-docker-host=unix:///var/run/docker.sock -insecure -sdk=%s -ds=%s -vm-path=%s %s",
+			conf.TargetPath, conf.ImageStorePath, vm.InventoryPath, vicadmintlsargs)})
+
+	// reconfig
+	info, err = tasks.WaitForResult(d.ctx, func(ctx context.Context) (tasks.ResultWaiter, error) {
+		return vm.Reconfigure(ctx, *spec)
+	})
+
+	if err != nil {
+		log.Errorf("Error while setting component parameters to appliance: %s", err)
+		return err
+	}
+	if info.State != types.TaskInfoStateSuccess {
+		log.Errorf("Setting parameters to appliance reported: %s", info.Error.LocalizedMessage)
+		return err
+	}
+
+	d.appliance = vm
+	return nil
+}
+
+func (d *Dispatcher) setMacToGuestInfo() error {
+	m, err := d.appliance.MacAddresses(d.ctx)
+	if err != nil || len(m) == 0 {
+		err = errors.Errorf("Failed to get VM mac address: %s", err)
+		return err
+	}
+	var spec types.VirtualMachineConfigSpec
+	spec = types.VirtualMachineConfigSpec{
+		ExtraConfig: []types.BaseOptionValue{},
+	}
+	var keys []string
+	for key, value := range m {
+		spec.ExtraConfig = append(spec.ExtraConfig, &types.OptionValue{Key: fmt.Sprintf("guestinfo.vch/networks/%s", d.nics[key]), Value: value})
+		keys = append(keys, d.nics[key])
+	}
+	spec.ExtraConfig = append(spec.ExtraConfig, &types.OptionValue{Key: "guestinfo.vch/networks", Value: strings.Join(keys, " ")})
+
+	// reconfig
+	_, err = tasks.WaitForResult(d.ctx, func(ctx context.Context) (tasks.ResultWaiter, error) {
+		return d.appliance.Reconfigure(ctx, spec)
+	})
+
+	if err != nil {
+		log.Errorf("Error to set MacAddress into guestinfo: %s", err)
+		return err
+	}
+
+	return nil
+}
+
+func (d *Dispatcher) fetchGuestInfoOrTimeout(dul time.Duration) (map[string]string, error) {
```

Comment:
This is a misleading name - it's not timeout waiting for GuestInfo; it's a timeout waiting for clientIP.
We should be updating this eventually to use the vSphere events - no need to poll.